### DISCUSSION TITLE
spoof source address seems to be ignored

### DISCUSSION THREAD

**Discussion by jebbtr** [mitmproxy/mitmproxy#4914] [Q&A] [answered]
Hi,

I'm playing with mitmproxy transparent mode and want to try full transparency.  I added the parameter 
--set spoof_source_address=true
and while this doesn't give an error and the proxy works OK it doesn' seem to have made any difference, ie when I examine the traffic from the proxy with wireshark the source ip addtress is not that of the client.  I there a type in the parameter, or have I misunderstood something?

./mitmproxy --version
Mitmproxy: 7.0.2 binary
Python:    3.9.5
OpenSSL:   OpenSSL 1.1.1k  25 Mar 2021
Platform:  Linux-5.4.0-84-generic-x86_64-with-glibc2.27

**Comment by mhils** [comment, marked as answer]
I think we have removed support for this in 7.0, there is a note in [CHANGELOG.md](https://github.com/mitmproxy/mitmproxy/blob/main/CHANGELOG.md#16-july-2021-mitmproxy-70):

> Remove the following options: `http2_priority`, `relax_http_form_validation`, `upstream_bind_address`, `spoof_source_address`, and `stream_websockets`. If you depended on one of them please let us know. mitmproxy never phones home, which means we don't know how prominently these options were used. (@mhils)

Transferring `spoof_source_address` over to asyncio wasn't trivial, so I left it out as it's a relatively niche use case and I don't have a test setup to verify that it works. That being said, I'd be happy to merge a PR that adds it back! 😃 

For reference:
  - This is the old implementation: https://github.com/mitmproxy/mitmproxy/blob/0407ce3106d7106ae4a6db2b3c1340f54e77f550/mitmproxy/net/tcp.py#L431-L441
  - This is where we open connections now: https://github.com/mitmproxy/mitmproxy/blob/0904cc3f5f0f95f6b9c3a6829e02f08ebb5bf9e7/mitmproxy/proxy/server.py#L153
  `open_connection` accepts a socket object, so initializing that first and then passing it in should work.

**Reply by jebbtr** [reply]
Well that would explain it, thanks. Maybe I'll try an older version.  It might be nice to add something saying it's no longer supported.  If you try --spoof-source-address you get
--spoof-source-address is deprecated.
Please use `--set spoof_source_address=value` instead.
To show all options and their default values use --options

Which is misleading if it's no longer supported.

**Reply by mhils** [reply]
Done in 7be646f, thanks!
